starlite/controller.py

```diff
@@ -67,6 +75,9 @@ def __init__(self, owner: "Router"):
 
         self.path = normalize_path(self.path or "/")
         self.owner = owner
+        self._after_request = AfterRequestHook(ensure_unbound(self.after_request)) if self.after_request else None
```

Comment:
> Why have this logic in s class var

Not class vars, instance vars, just like we do for app, router and handler.

>  rather than simply do this in the resolve method on the handler

I'll have a look at moving the logic like you suggest next time I sit down to this.

> Otherwise, why do you need the private variables and not simply reasign the variables.

Then we'd have to change the type definitions of the handlers on the `Controller` class to support the assignment, which would leak the implementation detail out.